File: dart_sample/lib/container_demo_page.dart
import 'package:flutter/material.dart';

import 'counter_widgets.dart';

class ContainerDemoPage extends StatefulWidget {
  const ContainerDemoPage({super.key});

  @override
  State<ContainerDemoPage> createState() => _ContainerDemoPageState();
}

class _ContainerDemoPageState extends State<ContainerDemoPage> {
  Alignment _alignment = Alignment.center;
  bool _withMargin = false;

  @override
  Widget build(BuildContext context) {
    final EdgeInsetsGeometry? margin = _withMargin
        ? const EdgeInsets.fromLTRB(12, 8, 12, 8)
        : null;

    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 10,
      children: <Widget>[
        const Text(
          'Container alignment + margin',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Switch alignment and margin to verify Container composition behavior.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Row(
          spacing: 8,
          children: <Widget>[
            _buildButton(
              label: 'TopLeft',
              onTap: () => _setAlignment(Alignment.topLeft),
              width: 96,
              background: const Color(0xFFDCE3ED),
            ),
            _buildButton(
              label: 'Center',
              onTap: () => _setAlignment(Alignment.center),
              width: 96,
              background: const Color(0xFFDCE3ED),
            ),
            _buildButton(
              label: 'BottomRight',
              onTap: () => _setAlignment(Alignment.bottomRight),
              width: 112,
              background: const Color(0xFFDCE3ED),
            ),
          ],
        ),
        Row(
          spacing: 8,
          children: <Widget>[
            _buildButton(
              label: _withMargin ? 'Margin: on' : 'Margin: off',
              onTap: _toggleMargin,
              width: 120,
              background: const Color(0xFFE9F5EC),
            ),
          ],
        ),
        Text(
          'alignment=${_alignmentLabel(_alignment)}, margin=${_withMargin ? 'on' : 'off'}',
          style: const TextStyle(fontSize: 12, color: Colors.blueGrey),
        ),
        Container(
          width: 240,
          height: 160,
          color: const Color(0xFFE7EDF6),
          padding: const EdgeInsets.all(8),
          child: Container(
            margin: margin,
            width: 150,
            height: 90,
            alignment: _alignment,
            decoration: BoxDecoration(
              color: const Color(0xFFCCE3FF),
              border: Border.all(color: const Color(0xFF1D3557), width: 2),
              borderRadius: BorderRadius.circular(12),
            ),
            child: Container(
              width: 32,
              height: 20,
              color: const Color(0xFF1D3557),
              child: const Center(
                child: Text(
                  'C',
                  style: TextStyle(fontSize: 12, color: Colors.white),
                ),
              ),
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildButton({
    required String label,
    required VoidCallback onTap,
    required double width,
    required Color background,
  }) {
    return SizedBox(
      width: width,
      child: CounterTapButton(
        label: label,
        onTap: onTap,
        background: background,
        foreground: Colors.black,
        fontSize: 12,
        padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 8),
      ),
    );
  }

  void _setAlignment(Alignment alignment) {
    setState(() {
      _alignment = alignment;
    });
  }

  void _toggleMargin() {
    setState(() {
      _withMargin = !_withMargin;
    });
  }

  static String _alignmentLabel(Alignment alignment) {
    if (alignment == Alignment.topLeft) {
      return 'topLeft';
    }

    if (alignment == Alignment.center) {
      return 'center';
    }

    if (alignment == Alignment.bottomRight) {
      return 'bottomRight';
    }

    return '(${alignment.x.toStringAsFixed(2)},${alignment.y.toStringAsFixed(2)})';
  }
}

File: src/Sample/Flutter.Net/ContainerDemoPage.cs
using System;
using Avalonia;
using Avalonia.Media;
using Flutter.Rendering;
using Flutter.Widgets;

// Dart parity source (reference): dart_sample/lib/container_demo_page.dart (exact sample parity)

namespace Flutter.Net;

public sealed class ContainerDemoPage : StatefulWidget
{
    public override State CreateState()
    {
        return new ContainerDemoPageState();
    }
}

internal sealed class ContainerDemoPageState : State
{
    private Alignment _alignment = Alignment.Center;
    private bool _withMargin;

    public override Widget Build(BuildContext context)
    {
        var margin = _withMargin ? new Thickness(12, 8, 12, 8) : (Thickness?)null;

        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 10,
            children:
            [
                new Text("Container alignment + margin", fontSize: 20, color: Colors.Black),
                new Text(
                    "Switch alignment and margin to verify Container composition behavior.",
                    fontSize: 14,
                    color: Colors.DimGray),
                new Row(
                    spacing: 8,
                    children:
                    [
                        BuildButton("TopLeft", () => SetAlignment(Alignment.TopLeft), width: 96, colorHex: "#FFDCE3ED"),
                        BuildButton("Center", () => SetAlignment(Alignment.Center), width: 96, colorHex: "#FFDCE3ED"),
                        BuildButton("BottomRight", () => SetAlignment(Alignment.BottomRight), width: 112, colorHex: "#FFDCE3ED"),
                    ]),
                new Row(
                    spacing: 8,
                    children:
                    [
                        BuildButton(_withMargin ? "Margin: on" : "Margin: off", ToggleMargin, width: 120, colorHex: "#FFE9F5EC"),
                    ]),
                new Text(
                    $"alignment={AlignmentLabel(_alignment)}, margin={(_withMargin ? "on" : "off")}",
                    fontSize: 12,
                    color: Colors.DarkSlateGray),
                new Container(
                    width: 240,
                    height: 160,
                    color: Color.Parse("#FFE7EDF6"),
                    padding: new Thickness(8),
                    child: new Container(
                        margin: margin,
                        width: 150,
                        height: 90,
                        alignment: _alignment,
                        decoration: new BoxDecoration(
                            Color: Color.Parse("#FFCCE3FF"),
                            Border: new BorderSide(Color.Parse("#FF1D3557"), 2),
                            BorderRadius: BorderRadius.Circular(12)),
                        child: new Container(
                            width: 32,
                            height: 20,
                            color: Color.Parse("#FF1D3557"),
                            child: new Center(
                                child: new Text("C", fontSize: 12, color: Colors.White))))),
            ]);
    }

    private Widget BuildButton(string label, Action onTap, double width, string colorHex)
    {
        return new SizedBox(
            width: width,
            child: new CounterTapButton(
                label: label,
                onTap: onTap,
                background: Color.Parse(colorHex),
                foreground: Colors.Black,
                fontSize: 12,
                padding: new Thickness(10, 8)));
    }

    private void SetAlignment(Alignment alignment)
    {
        SetState(() => _alignment = alignment);
    }

    private void ToggleMargin()
    {
        SetState(() => _withMargin = !_withMargin);
    }

    private static string AlignmentLabel(Alignment alignment)
    {
        if (alignment == Alignment.TopLeft)
        {
            return "topLeft";
        }

        if (alignment == Alignment.Center)
        {
            return "center";
        }

        if (alignment == Alignment.BottomRight)
        {
            return "bottomRight";
        }

        return $"({alignment.X:0.##},{alignment.Y:0.##})";
    }
}
